### PR TITLE
Rename rookpool and rookcluster to pool and cluster

### LINKED ISSUE #580: Rename Rookcluster back to Cluster to avoid stutter and use beta TPRs ASAP
The TPR to beta proposal is here: https://github.com/kubernetes/community/pull/524
It will allow us to customize the kubectl names easily without having to have an unique resource name cluster-wide.

Therefore I propose to rename things back to Cluster, Pool, etc.
In the meantime it's possible to do `kubectl get clusters.rook.io` which I think is reasonable (this is only given the corner case that the etcd operator also has the Cluster TPR registered)

### PR DESCRIPTION
**Pull request by travisn**
Fixes #580

### CODE REVIEW

File: Documentation/kubernetes.md [hunk 1]
```diff
@@ -80,8 +80,9 @@ To learn how to set up monitoring for your Rook cluster, you can follow the step
 To clean up all the artifacts created by the demo, **first cleanup the resources from the block, file, and object walkthroughs** (unmount volumes, delete volume claims, etc), then run the following:
 ```bash
 kubectl delete deployment rook-operator
-kubectl delete -n rook rookcluster rook
-kubectl delete thirdpartyresources rookcluster.rook.io rookpool.rook.io
+kubectl delete -n rook cluster rook
+kubectl delete thirdpartyresources cluster.rook.io pool.rook.io
+kubectl delete storageclass rook-block
```
Comment (kokhang) [inline Documentation/kubernetes.md]: The deletion of the storageclass is already documented in https://github.com/rook/rook/blob/master/Documentation/k8s-block.md#teardown

Comment (travisn) [inline Documentation/kubernetes.md]: oops, missed that when resolving the merge conflict

**Comment by kokhang** [comment]
@travisn #580 states to use TPR beta. Should we create a new issue just for this?

**Comment by jbw976** [comment]
@kokhang it looks like that PR to move TPR to beta is still open, I don't think we can make that move yet: https://github.com/kubernetes/community/pull/524

**Comment by luxas** [comment]
Yeah, beta TPRs can't be used yet, they will probably come with v1.7

**Pull request merged**